### PR TITLE
Added templates

### PR DESCRIPTION
**Pull request by amarozinfoblox**
Added ARM templates.

### CODE REVIEW

File: main/createUiDefinition.json [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        "name": "adminPassword",
+        "toolTip": "Password for the user",
+        "type": "Microsoft.Common.PasswordBox",
+        "label": {
+          "password": "Password",
+          "confirmPassword": "Confirm password"
+        },
+        "constraints": {
+          "required": true,
+          "regex": "^(?:(?=.*[a-z])(?:(?=.*[A-Z])(?=.*[\\d\\W])|(?=.*\\W)(?=.*\\d))|(?=.*\\W)(?=.*[A-Z])(?=.*\\d)).{12,64}$",
```
Comment (jkraj) [inline main/createUiDefinition.json:31]: This is not needed. lets hardcode the password under `osprofile` anyway we are not using it.

File: main/nic-pip-existing.json [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      "name": "[parameters('nicName')]",
+      "location": "[parameters('location')]",
+      "properties": {
+        "ipConfigurations": [
+          {
+            "name": "ipconfig1",
+            "properties": {
+              "primary": true,
+              "privateIPAllocationMethod": "Dynamic",
+              "privateIPAddressVersion": "IPv4",
```
Comment (jkraj) [inline main/nic-pip-existing.json:34]: do AWS support IPV6?

File: main/publicip-new.json [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    "publicIpAddressName": {
+      "type": "string"
+    },
+    "publicIpAddressType": {
+      "type": "string"
+    }
+  },
+  "resources": [
+    {
+      "apiVersion": "2020-08-01",
```
Comment (jkraj) [inline main/publicip-new.json:17]: why this date? this version can't be in vX.Y.Z format?

File: main/vm.json [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      "hardwareProfile": "[parameters('hardwareProfile')]",
+      "osProfile": "[parameters('osProfile')]",
+      "storageProfile": "[parameters('storageProfile')]",
+      "networkProfile": "[parameters('networkProfile')]",
+      "diagnosticsProfile": "[parameters('diagnosticsProfile')]"
+    }
+  },
+  "resources": [
+    {
+      "apiVersion": "2020-12-01",
```
Comment (jkraj) [inline main/vm.json:41]: why this date?

File: main/createUiDefinition.json [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+          "validationMessage": "The password must be between 12 and 64 characters long, and contain characters from at least 3 of the following groups: uppercase characters, lowercase characters, numbers, and special characters."
+        },
+        "visible": true
+      },
+      {
+        "name": "customData",
+        "type": "Microsoft.Common.TextBox",
+        "label": "Custom data",
+        "toolTip": "Pass custom data configuration.",
```
Comment (jkraj) [inline main/createUiDefinition.json:40]: what is the difference between custom-data vs cloud-init `userdata` but we are passing it right?

**Review by jkraj — COMMENTED** [review]
VM doesn't need any security policy to enable access? say 443 port for bootstrap UI

File: main/mainTemplate.json [hunk 6]
```diff
@@ -0,0 +1,372 @@
...
+  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "parameters": {
+    "artifactsBaseUrl": {
+      "type": "string",
+      "metadata": {
+        "artifactsBaseUrl": "/",
+        "description": "The base URL for dependent assets."
+      },
+      "defaultValue": "https://raw.githubusercontent.com/amarozinfoblox/azure-templates/master/main/"
```
Comment (nnallamo) [inline main/mainTemplate.json:11]: do we need to refer infobloxopen repo here ?

File: main/createUiDefinition.json (hunk 5 above)
Comment (amarozinfoblox) [reply main/createUiDefinition.json:25]: Azure supports two formats: `User data` and `Custom data`. As written in the documentation:
- [User data](https://docs.microsoft.com/en-us/azure/virtual-machines/user-data) is a new version of [custom data](https://docs.microsoft.com/en-us/azure/virtual-machines/custom-data)

And yes we are passing our userdata to Azure customData and it works fine. Our userdata does not work with new version `User data` of Azure.

File: main/nic-pip-existing.json (hunk 2 above)
Comment (amarozinfoblox) [reply main/nic-pip-existing.json:34]: In arm template we can configure IPv6, but according to Azure certificate tool we must use IPv4.

File: main/publicip-new.json (hunk 3 above)
Comment (amarozinfoblox) [reply main/publicip-new.json:17]: This is format of Azure.

File: main/vm.json (hunk 4 above)
Comment (amarozinfoblox) [reply main/vm.json:41]: This is format of Azure.

**Comment by amarozinfoblox** [comment]
> VM doesn't need any security policy to enable access? say 443 port for bootstrap UI

Added rule for 443 port.

File: main/mainTemplate.json (hunk 6 above)
Comment (amarozinfoblox) [reply main/mainTemplate.json]: Changed to correct url back.

**Review by nnallamo — APPROVED** [review]

File: main/vm.json (hunk 4 above)
Comment (jkraj) [reply main/vm.json:41]: Format is fine why it is 2020? Do you copied from NIOS template?

File: main/nic-pip-existing.json (hunk 2 above)
Comment (jkraj) [reply main/nic-pip-existing.json:34]: You mean we should use only IPV4 not both 4 & 5. I mean we should add a option to configure IPV6 as well. May be in next PR

File: main/nic-pip-none.json [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+              "primary": true,
+              "privateIPAllocationMethod": "Dynamic",
+              "privateIPAddressVersion": "IPv4",
+              "subnet": {
+                "id": "[parameters('subnetId')]"
+              }
+            }
+          }
+        ],
+        "networkSecurityGroup": {
```
Comment (jkraj) [inline main/nic-pip-none.json:38]: Do we have option to use existing security group?

Comment (amarozinfoblox) [reply main/nic-pip-none.json:38]: No, we are creating new security group during deploy.

File: main/nic-pip-existing.json (hunk 2 above)
Comment (amarozinfoblox) [reply main/nic-pip-existing.json:34]: OK.

File: main/vm.json (hunk 4 above)
Comment (amarozinfoblox) [reply main/vm.json]: Updated all apiVersion fields to latest.

**Review by jkraj — APPROVED** [review]

**Pull request merged**